### PR TITLE
Resize term on event

### PR DESCRIPTION
**Pull request by hall**
**What this PR does / why we need it**:
These changes allow the pty to be resized if the frontend sends a specific string over the socket.

I read about `SIGWINCH` and `TIOCGWINSZ` but didn't see if there was a proper way of passing them between `xtermjs` and `crypto/ssh`. Instead, this intercepts the equivalent ANSI CSI sequence (`\u001b[8;<H>;<W>t`).

I also removed the patch from the go version in go.mod due to the following error:
```
$ go build
go: errors parsing go.mod:
/home/bryton/src/github.com/hobbyfarm/gargantua/go.mod:3: usage: go 1.23
```
**Which issue(s) this PR fixes**:

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/hobbyfarm/gargantua
 
-go 1.13.4
+go 1.13
```
Comment (Oats87) [inline go.mod:3]: Can we move this into its own PR?

File: .gitignore [hunk 2]
```diff
@@ -16,3 +16,7 @@ artifacts/testing
 
 # docker-compose enviornment variables
 /.env
+
+# local dev logs
+garg
```
Comment (Oats87) [inline .gitignore:21]: Can we move this into its own PR?

**Comment by Oats87** [comment]
@hall thanks for the updated PR, I am also going to ask @luthermonson to take a look at this as well for a second set of eyes

File: pkg/shell/shell.go [hunk 3]
```diff
@@ -188,7 +182,7 @@ func (sp ShellProxy) ConnectFunc(w http.ResponseWriter, r *http.Request) {
 		io.Copy(pip, stdin)
 	}()
 
-	err = sess.RequestPty("xterm", 40, width, ssh.TerminalModes{ssh.ECHO: 1, ssh.TTY_OP_ISPEED: 14400, ssh.TTY_OP_OSPEED: 14400})
+	err = sess.RequestPty("xterm", 40, 80, ssh.TerminalModes{ssh.ECHO: 1, ssh.TTY_OP_ISPEED: 14400, ssh.TTY_OP_OSPEED: 14400})
```
Comment (Oats87) [inline pkg/shell/shell.go:185]: Is the general idea here that we start out a session at `40x80` and then `SIGWINCH` it to the correct size? Why not create the terminal session at the initial size we know we want, then only on resize change it via `SIGWINCH`?

File: pkg/shell/shell.go [hunk 4]
```diff
@@ -39,6 +43,7 @@ func init() {
 	sshDev = os.Getenv("SSH_DEV")
 	sshDevHost = os.Getenv("SSH_DEV_HOST")
 	sshDevPort = os.Getenv("SSH_DEV_PORT")
+	SIGWINCH = regexp.MustCompile(`.*\[8;(.*);(.*)t`)
```
Comment (Oats87) [inline pkg/shell/shell.go:46]: Where did you source this regex from?

File: pkg/shell/shell.go (hunk 3 above)
Comment (hall) [reply pkg/shell/shell.go:185]: Yep -- that's exactly the idea. I think I got confused and thought `width` was on my branch (I didn't realize the UI was already sending that data). The corresponding UI code for this is sending a resize signal on initialization but, to your point, we can remove that step and set the correct size the first time.

File: pkg/shell/shell.go (hunk 4 above)
Comment (hall) [reply pkg/shell/shell.go:46]: I pieced it together from a few different webpages (e.g., lists of CSI codes and the xterm.js docs). The general idea is captured here: https://stackoverflow.com/questions/45775005/xterm-js-ws-ssh2-not-transmitting-terminal-resize-signals-sigwinch-to-ssh

**Comment by ebauman** [comment]
Closing this in favor of #81
